File: easy/3014-minimum-number-of-pushes-to-type-word-i/solution.java
/**
 * 3014. Minimum Number of Pushes to Type Word I
 * Time: O(n)
 * Space: O(1)
 */
class Solution {
    public int minimumPushes(String word) {
        int total = 0;
        // Distinct letters: the i-th letter (0-based) costs i / 8 + 1 pushes.
        for (int i = 0; i < word.length(); i++) {
            total += i / 8 + 1;
        }
        return total;
    }
}

File: easy/3014-minimum-number-of-pushes-to-type-word-i/solution.py
class Solution:
    def minimumPushes(self, word: str) -> int:
        """
        3014. Minimum Number of Pushes to Type Word I
        Time: O(n)
        Space: O(1)
        """
        # Distinct letters: the i-th letter (0-based) costs i // 8 + 1 pushes.
        return sum(i // 8 + 1 for i in range(len(word)))
